File: src/lib/weather.ts
const API_KEY = process.env.OPENWEATHER_API_KEY;
const BASE_URL = 'https://api.openweathermap.org/data/2.5/weather';

export interface WeatherData {
    condition: string; // 'Rain', 'Clear', 'Clouds', etc.
    description: string;
    temp: number; // Celsius
    isRaining: boolean;
}

export async function getWeather(lat: number, lon: number): Promise<WeatherData | null> {
    if (!API_KEY) {
        console.warn("OPENWEATHER_API_KEY is not set. Returning mock weather data.");
        // Mock data for development/demo if key is missing
        return {
            condition: 'Clear',
            description: 'clear sky',
            temp: 22,
            isRaining: false
        };
    }

    try {
        const response = await fetch(`${BASE_URL}?lat=${lat}&lon=${lon}&units=metric&appid=${API_KEY}`);
        if (!response.ok) {
            throw new Error(`Weather API error: ${response.statusText}`);
        }
        const data = await response.json();

        const condition = data.weather[0]?.main || 'Clear';

        return {
            condition,
            description: data.weather[0]?.description || '',
            temp: data.main.temp,
            isRaining: condition === 'Rain' || condition === 'Drizzle' || condition === 'Thunderstorm'
        };

    } catch (error) {
        console.error("Failed to fetch weather:", error);
        return null;
    }
}

File: python_app/utils/weather.py
import os
import requests
from dotenv import load_dotenv

load_dotenv('.env.local')

OPENWEATHER_API_KEY = os.environ.get("OPENWEATHER_API_KEY")

def get_weather(lat: float, lon: float):
    """
    Fetches current weather data for the given coordinates.
    Returns a dict with condition, description, temp, and is_raining boolean.
    """
    if not OPENWEATHER_API_KEY:
        return {
            "condition": "Clear",
            "description": "Clear sky (Mock)",
            "temp": 25.0,
            "is_raining": False
        }

    try:
        url = f"https://api.openweathermap.org/data/2.5/weather?lat={lat}&lon={lon}&units=metric&appid={OPENWEATHER_API_KEY}"
        response = requests.get(url, timeout=5)
        response.raise_for_status()
        data = response.json()

        condition = data['weather'][0]['main']
        description = data['weather'][0]['description']
        temp = data['main']['temp']
        
        is_raining = condition in ['Rain', 'Drizzle', 'Thunderstorm']

        return {
            "condition": condition,
            "description": description,
            "temp": temp,
            "is_raining": is_raining
        }
    except Exception as e:
        print(f"Error fetching weather: {e}")
        return {
            "condition": "Unknown",
            "description": "Unavailable",
            "temp": 20.0,
            "is_raining": False
        }
